src/promise_provider.ts

```diff
@@ -4,11 +4,11 @@ import { MongoInvalidArgumentError } from './error';
 const kPromise = Symbol('promise');
 
 interface PromiseStore {
-  [kPromise]?: PromiseConstructor;
+  [kPromise]: PromiseConstructor | null;
```

Comment:
haven't reviewed this file yet - pending team discussion about breaking changes here